src/modules/recovery/main_impl.h

```diff
@@ -120,6 +120,69 @@ static int secp256k1_ecdsa_sig_recover(const secp256k1_ecmult_context *ctx, cons
     return !secp256k1_gej_is_infinity(&qj);
 }
 
+static int secp256k1_ecdsa_sig_recover_four(const secp256k1_ecmult_context *ctx, secp256k1_gej out[], size_t *n, const secp256k1_scalar *sigr, const secp256k1_scalar *rn, const secp256k1_scalar* sigs, const secp256k1_scalar *message) {
```

Comment:
> secp256k1_gej out[]

FWIW, you pretty much never want to pass an sized array argument in C.  ( https://lkml.org/lkml/2015/9/3/428 )

Reply:
Yeah I know. I did it because I find it more readable than const pointers of const pointers.
But if it bothers people i'll refactor the function declarations before it get's out of WIP.

Reply:
Actually I'm a bit sad that because of stupidity of people doing `sizeof(arr)` and expecting to get the size of the arrays we lose array arguments.
I find them as a nice way for the code to self document the function's contract.